frontend/taipy-gui/src/components/Taipy/Metric.spec.tsx

```diff
@@ -11,21 +11,415 @@
+                [0.3333333333333333, "#a1d99b"],
+                [0.4444444444444444, "#74c476"],
+                [0.5555555555555556, "#41ab5d"],
+                [0.6666666666666666, "#238b45"],
+                [0.7777777777777778, "#006d2c"],
+                [0.8888888888888888, "#00441b"],
+                [1, "#000000"]
+            ]
+        },
+        paper_bgcolor: "rgb(255,255,255)",
+    }
+}
+
 describe("Metric Component", () => {
-  it("renders", async () => {
-    const { getByTestId } = render(<Metric testId="test-id"/>);
-    const elt = getByTestId("test-id");
-    expect(elt.tagName).toBe("DIV");
-  })
-  it("displays the right info for class", async () => {
-    const { getByTestId } = render(<Metric testId="test-id" className={'taipy-gauge'}/>);
-    const elt = getByTestId('test-id');
-    expect(elt).toHaveClass('taipy-gauge');
-  })
+    it("renders", async () => {
+        const {getByTestId} = render(<Metric testId="test-id"/>);
+        const elt = getByTestId("test-id");
+        expect(elt.tagName).toBe("DIV");
+    })
+
+    it("displays the right info for class", async () => {
+        const {getByTestId} = render(<Metric testId="test-id" className={'taipy-gauge'}/>);
+        const elt = getByTestId('test-id');
+        expect(elt).toHaveClass('taipy-gauge');
+    })
+
+    it("sets the title when provided", async () => {
+        const title = "Test Title";
+        let container: RenderResult | null = null;
+        container = render(<Metric title={title}/>);
+        await waitFor(() => {
+            const titleElement = container!.container.querySelector('.gtitle');
+            if (!titleElement) {
+                throw new Error('Title element not found');
+            }
+            expect(titleElement.textContent).toContain(title);
+        });
+    });
+
+    it('logs an error when template prop is a malformed JSON string', () => {
+        const consoleSpy = jest.spyOn(console, 'info');
+        const malformedJson = "{ key: 'value'";
+        render(<Metric template={malformedJson}/>);
+        expect(consoleSpy).toHaveBeenCalledWith(expect.stringContaining('Error while parsing Metric.template'));
+        consoleSpy.mockRestore();
+    });
+
+    it('logs an error when colorMap prop is a malformed JSON string', () => {
+        const consoleSpy = jest.spyOn(console, 'info');
+        const malformedJson = "{ key: 'value'"; // missing closing brace
+        render(<Metric colorMap={malformedJson}/>);
+        expect(consoleSpy).toHaveBeenCalledWith(expect.stringContaining('Error parsing color_map value (metric)'));
+        consoleSpy.mockRestore();
+    });
+
+    it("sets the template when provided", async () => {
+        let container: RenderResult | null = null;
+        container = render(<Metric template={JSON.stringify(template)}/>);
+        await waitFor(() => {
+            const elmWithTpl = container!.container.querySelector('.user-select-none')
```

Comment:
How does this test the use of the template ?

Reply:
I've refactored this test case to check the use of template